### ISSUE TITLE
Kitt4SME Demo Cluster deployment

### ISSUE THREAD

**Issue by cmelchiorre** [c0c0n3/kitt4sme.live#127]
Not sure this is the right place to post this type of technical question: We’re the responsibles for the Open Call action TA08 AIGreenWaste. 

We’re trying to install a demo Kitt4SME cluster following the instructions at the link
https://github.com/c0c0n3/kitt4sme/blob/master/poc/deployment/install.md
as suggested by Elena Minisci in view of the incoming technical meeting in mid-April.

Unfortunately, we are having problem in following the steps in the above page. 
When arrived at the istioctl install step, this failed with a series of errors (please see the attached files).

Attached you’ll find:

- provision history.sh with the list of commands executed up to the failed step

[provision history.sh.txt](https://github.com/c0c0n3/kitt4sme.live/files/8404251/provision.history.sh.txt)

- versions.txt the version of the installed components (docker, minikube, kubectl, istio)

[versions.txt](https://github.com/c0c0n3/kitt4sme.live/files/8404252/versions.txt)

- istioctl install error.txt the log of the error 

[istioctl install error.txt](https://github.com/c0c0n3/kitt4sme.live/files/8404253/istioctl.install.error.txt)

Could you tell us if we’re following all the steps right?

Manu thanks,

**Comment by berserkrambo** [comment]
Hi, I'm Riccardo from GoatAI, AI4SWD

I had the same problem and I solved it installing exactly the same version described in the readme.md.
If I am correct you installed minikube via apt-install, you should install this one: https://github.com/kubernetes/minikube/releases/download/v1.15.1/minikube_1.15.1-0_amd64.deb
One more thing, to use kubectl you don''t have to install it, you have it already, just run "minikube kubectl" and not only "kubectl".

BUT, solving this one is not enough, that same line of code:
    "istioctl install -f poc/deployment/manual/istio/demo-profile.yaml -y --verify"
doesn't work anyway xD
giving us:
Detected that your cluster does not support third party JWT authentication. Falling back to less secure first party JWT. See https://istio.io/v1.9/docs/ops/best-practices/security/#configure-third-party-service-account-tokens for details.
✔ Istio core installed                                                                                                                         
✘ Istiod encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition


Hope I helped you, and hope someone help will me/us on this :)

Bye

**Comment by cmelchiorre** [comment]
Thanks Riccardo: I confirm that by using the new version  my problem is solved.

“unfortunately” I’m not getting the other error you mentioned  (meaning, I can’t help with this), but instead this brought me up to almost the end of the tutorial steps.

Now, I’m stuck at the steps "argocd app create kitt4sme", where I got the following errors - when launching;

xdg-open http://$(minikube ip -p kitt4sme):8080
I get:
/usr/bin/xdg-open: 778: /usr/bin/xdg-open: www-browser: not found
/usr/bin/xdg-open: 778: /usr/bin/xdg-open: links2: not found
/usr/bin/xdg-open: 778: /usr/bin/xdg-open: elinks: not found
/usr/bin/xdg-open: 778: /usr/bin/xdg-open: links: not found
/usr/bin/xdg-open: 778: /usr/bin/xdg-open: lynx: not found
/usr/bin/xdg-open: 778: /usr/bin/xdg-open: w3m: not found
xdg-open: no method available for opening 'http://192.168.49.2:8080'

and the "argocd app create kitt4sme" instruction itself ends with the following FATAL error:

FATA[0017] rpc error: code = InvalidArgument desc = application spec is invalid: InvalidSpecError: repository not accessible: Get "https://github.com/c0c0n3/kitt4sme/info/refs?service=git-upload-pack": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

**Comment by berserkrambo** [comment]
Finally found the solution to "my problem", just follow this other guide and all will work perfectly lol https://github.com/c0c0n3/kitt4sme.live/blob/main/docs/bootstrap.md

**Comment by cmelchiorre** [comment]
Thanks, Riccardo, I also followed the other guide and managed to install the microk8s cluster.

With one issue, though, since the guide mentions in the "Post-install steps" that "The only unhappy service should be Keycloak" which pod should show a CreateContainerConfigError. In my case the keycloak pod doesn't even appear in the list of the pods returned by "kubectl get pods --all-namespaces" command. I hope this doesn't mean I'm skipping some important step; I searched again through the guide but found no previous mention of keycloak...
